nixos/modules/services/web-servers/unit/default.nix

```diff
@@ -0,0 +1,70 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.unit;
+
+  configFile = builtins.toFile "unit.json" ''
+    ${cfg.config}
+  '';
+
+in {
+  options = {
+    services.unit = {
+      enable = mkEnableOption "Unit App Server";
+      package = mkOption {
+        type = types.package;
+        default = pkgs.unit;
+        description = "Unit package to use.";
+      };
+      stateDir = mkOption {
+        default = "/var/spool/unit";
+        description = "Unit data directory.";
+      };
+      logDir = mkOption {
+        default = "/var/log/unit";
+        description = "Unit log directory.";
+      };
+      config = mkOption {
+        type = types.lines;
+        default = ''
+          {
+            "listeners": {},
+            "applications": {}
+          }
+        '';
+        description = "Unit configuration in JSON format.";
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+
+    environment.systemPackages = [ cfg.package ];
+
+    systemd.tmpfiles.rules = [
+      "d ${cfg.stateDir} 0755 root root - -"
+      "d ${cfg.logDir} 0755 root root - -"
+     ];
+
+    systemd.services.unit = {
+      description = "Unit App Server";
+      after = [ "network.target" ];
+      wantedBy = [ "multi-user.target" ];
+      path = with pkgs; [ curl ];
+      preStart = ''
+        test -f /run/unit/control.unit.sock || rm -f /run/unit/control.unit.sock
+      '';
+      postStart = ''
+        curl -X PUT --data-binary @${configFile} --unix-socket /run/unit/control.unit.sock http://localhost/config
+      '';
+      serviceConfig = {
+        Type = "simple";
+        PIDFile = "/run/unit/unit.pid";
```

Comment:
`unitd` apparently has a `--no-daemon` option. Running in the foreground is preferred in NixOS, so you should use that instead (-> no need for pid file)